src/libs/PolicyUtils.ts

```diff
@@ -153,49 +153,57 @@ function getIneligibleInvitees(policyMembers: OnyxEntry<PolicyMembers>, personal
 }
 
 /**
- * Gets the tag from policy tags, defaults to the first if no key is provided.
+ * Gets a tag name of policy tags based on a tag index.
  */
-function getTag(policyTags: OnyxEntry<PolicyTags>, tagKey?: keyof typeof policyTags): PolicyTag | undefined | EmptyObject {
-    if (isEmptyObject(policyTags)) {
-        return {};
+function getTagListName(policyTagList: OnyxEntry<PolicyTagList>, tagIndex: number): string {
+    if (isEmptyObject(policyTagList)) {
+        return '';
     }
 
-    const policyTagKey = tagKey ?? Object.keys(policyTags ?? {})[0];
+    const policyTagKeys = Object.keys(policyTagList ?? {});
+    const policyTagKey = policyTagKeys[tagIndex] ?? '';
 
-    return policyTags?.[policyTagKey] ?? {};
+    return policyTagList?.[policyTagKey]?.name ?? '';
 }
 
 /**
- * Gets the first tag name from policy tags.
+ * Gets all tag lists of a policy
  */
-function getTagListName(policyTags: OnyxEntry<PolicyTags>) {
-    if (Object.keys(policyTags ?? {})?.length === 0) {
-        return '';
+function getTagLists(policyTagList: OnyxEntry<PolicyTagList>): Array<PolicyTagList[keyof PolicyTagList]> {
```

Comment:
Heads up that using indexed types in this manner [goes against our TS style guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/TS_STYLE.md#utility-types). 

Instead of `PolicyTagList[keyof PolicyTagList]`, we should use the much more readable `ValueOf` utility from `type-fest`:

```ts
function getTagLists(policyTagList: OnyxEntry<PolicyTagList>): Array<ValueOf<PolicyTagList>> {
```

I'm tempted to create an issue to create a lint rule to enforce this and automatically update it via Prettier, but we have more urgent matters at hand.

Reply:
Thank you for this comment. I will use `ValueOf` next time 🙂